python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -206,24 +208,104 @@ def __post_init__(self) -> None:
         if self.file_options.n_rows is not None:
             raise NotImplementedError("row limit in scan")
         if self.typ not in ("csv", "parquet"):
+            raise NotImplementedError(f"Unhandled scan type: {self.typ}")
+        if self.cloud_options is not None and any(
+            self.cloud_options[k] is not None for k in ("aws", "azure", "gcp")
```

Comment:
Confirming that these strings are guaranteed to be in the dict at this point? If this option isn't supported at all would it be worthwhile just to only accept `None` here?

Reply:
csv reader has non-None cloud_options but the values are None, so I can't punt for `cloud_options is not None`.